sdk/src/main/java/com/mapbox/maps/MapView.kt

```diff
@@ -34,7 +35,18 @@ import com.mapbox.maps.renderer.egl.EGLCore
  * and for ensuring your use adheres to the relevant terms of use.
  */
 open class MapView : FrameLayout, MapPluginProviderDelegate, MapControllable {
-  private var mapController: MapController
+  internal var mapController: MapController
+    private set
+
+  private var isViewAnnotationManagerInitialized = false
+  /**
+   * Get view annotation manager instance to add / update / remove view annotations
+   * represented as Android views.
+   */
+  val viewAnnotationManager: ViewAnnotationManager by lazy {
+    isViewAnnotationManagerInitialized = true
+    ViewAnnotationManagerImpl(this)
+  }
```

Comment:
Kotlin language feature of lazy don't require you to manage the initalization state yourself.
See https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-lazy/is-initialized.html

Reply:
I was afraid for some reason that calling this method when destroying map view will first create an instance and then destroy it but seems that it's not true, will apply your solution here.

Reply:
haha same, but you do need to use something as:

```
val messageBrokerDelegate = lazy { MessageBroker() }
val messageBroker by messageBrokerDelegate

if(messageBrokerDelegate.isInitialized())
    ...
```

from https://stackoverflow.com/questions/42522739/kotlin-check-if-lazy-val-has-been-initialised

Reply:
Applied, thanks, works like a charm!